webapp/src/ts/modules/contacts/contacts.component.ts

```diff
@@ -310,6 +310,52 @@ export class ContactsComponent implements OnInit, AfterViewInit, OnDestroy {
     return this.sortDirection === 'last_visited_date';
   }
 
+  private getContactsDocIds(contactsList, usersHomePlace): string[] {
+    const docIds = contactsList.map((item) => item._id);
+    if (usersHomePlace && !docIds.includes(usersHomePlace._id)) {
+      docIds.push(usersHomePlace._id);
+    }
+    return docIds;
+  }
+
+  private findHomeIndex(updatedContacts: any[]): number {
+    return _findIndex(updatedContacts, contact => contact._id === this.usersHomePlace._id);
+  }
+
+  private checkAdditionalListItem(homeIndex: number) {
+    this.additionalListItem =
+      !this.filters.search &&
+      (this.additionalListItem || !this.appending) &&
+      homeIndex === -1;
+  }
+
+  private moveUserHomePlaceToTop(updatedContacts: any[], homeIndex: number) {
+    if (homeIndex !== -1) {
+      this.usersHomePlace = updatedContacts[homeIndex];
+
+      if (!this.appending) {
+        return [this.usersHomePlace, ...updatedContacts.slice(1)];
+      }
+
+      if (this.additionalListItem) {
+        return [this.usersHomePlace, ...updatedContacts];
+      }
+
+      return updatedContacts;
+    }
+  }
```

Comment:
Nice! Whenever you can return early, the code will be flatter. Also, sonar will count the cognitive complexity of the following `if` statements as 1. 
When `if` statements are nested in a block, sonar counts them as 2, therefore increasing the score of cognitive complexity. 

```suggestion
  private moveUserHomePlaceToTop(updatedContacts: any[], homeIndex: number) {
    if (homeIndex < 0) {
      return;
    }
    
    this.usersHomePlace = updatedContacts[homeIndex];

    if (!this.appending) {
      return [this.usersHomePlace, ...updatedContacts.slice(1)];
    }

    if (this.additionalListItem) {
      return [this.usersHomePlace, ...updatedContacts];
    }

    return updatedContacts;
  }
```

Reply:
This means the function returns `undefined` and an array of records
I wonder if you can do `Record<string, string|number|bool>[]` instead of `any[]`

Reply:
Actually, should that first condition return the updated contact list? 
```
if (homeIndex < 0) {
  return updatedContacts;
}
```
I think it should, right?